packages/calypso-e2e/src/lib/pages/editor-page.ts

```diff
@@ -769,10 +769,13 @@ export class EditorPage {
 	 */
 	async unpublish(): Promise< void > {
 		const editorParent = await this.editor.parent();
-
 		await this.editorToolbarComponent.switchToDraft();
 		// @TODO: eventually refactor this out to a ConfirmationDialogComponent.
-		await editorParent.getByRole( 'button' ).getByText( 'OK' ).click();
+		// Saves the draft
+		await Promise.race( [
+			this.editorToolbarComponent.clickPublish(),
+			editorParent.getByRole( 'button' ).getByText( 'OK' ).click(),
```

Comment:
Why do we need both these calls? They seem to accomplish the same thing 🤔

The test also seems to complete successfully (on Simple and on Atomic) with only one of these lines.

Reply:
Actually the edge versions on simple and AT are the same and you won't notice a difference there. There are both running 18.2.0.

For versions before i.e. 18.1.2, which is what ordinary sites ( non-edge sites ) are using.

You can test this with:

`yarn jest specs/editor/editor__post-advanced-flow.ts`

... without the `GUTENBERG_EDGE=1` or  `TEST_ON_ATOMIC=true`. Apologies for not leaving this in the description.

The UI for that version is different. That OK button is only present there.

<img width="950" alt="Screenshot 2024-05-06 at 10 48 09" src="https://github.com/Automattic/wp-calypso/assets/277661/e048574b-3112-45ee-964c-a16347bcfb5a">

Without using `Promise.race()` with those two lines, you'll get this error:

```

  ● Editor: Advanced Post Flow › Revert post to draft › Switch to draft

    locator.click: Timeout 10000ms exceeded.
    =========================== logs ===========================
    waiting for frameLocator('iframe[src*="calypsoify"]').locator('body.block-editor-page').locator('.interface-navigable-region[class*="header"] button.editor-post-publish-button__button[aria-disabled="false"]')
    ============================================================

      276 | 		const editorParent = await this.editor.parent();
      277 | 		const publishButtonLocator = editorParent.locator( selectors.publishButton( 'enabled' ) );
    > 278 | 		await publishButtonLocator.click();
          | 		                           ^
      279 | 	}
      280 |
      281 | 	/**

      at EditorToolbarComponent.click [as clickPublish] (../../packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts:278:30)
      at EditorPage.unpublish (../../packages/calypso-e2e/src/lib/pages/editor-page.ts:775:3)
      at Object.<anonymous> (specs/editor/editor__post-advanced-flow.ts:155:4)


```

Reply:
Gotcha! I think I simply forgot to build after making my changes to the tests, or they wouldn't have passed 🤦‍♂️